### DISCUSSION TITLE
Cannot initialize data , please try again postgresql 16.

### DISCUSSION THREAD

**Discussion by rehanzproject** [leokhoa/laragon#818] [Q&A] [answered]
![image](https://github.com/leokhoa/laragon/assets/106218259/a8be737d-92f5-4e28-93de-812c44664349)
i've installed postgresql-16 on my laragon , but the laragon cannot initialized data , i don't know why it doesn't work.
i also tried at version 14 , and version 13 but still not working. 

Help me guyss

**Comment by Pen-y-Fan** [comment]
It is probably something on your computer preventing the directory from being created. It could be file permissions, anti-virus, Windows updates, locks to files or many other things.

Reboot your PC to release file locks, check for Windows updates and install them.

For permissions see my blog: [How to set up Laragon on a new Windows computer (part 2) - permissions](https://pen-y-fan.github.io/2022/11/08/how-to-set-up-laragon-on-a-new-windows-computer-part-2/).

Try to create the directory D:\laragon\data\postgresql-16 and add a text file to the directory.

If that doesn't fix the issue, view anti-virus logs for programs being prevented from running, directories created etc.

**Comment by bayukartiko** [comment, marked as answer]
That is an extraction issue when you download from Laragon "Tools" > "Quick Add"

Try manually downloading from https://www.enterprisedb.com/download-postgresql-binaries

Then extract the zip into C:\laragon\bin\postgresql

Don't forget to set the PostgreSQL directory in Laragon, pointing to the folder name that you just extracted
![image](https://github.com/user-attachments/assets/72d4cd2a-7cf9-4455-bd61-c63e00671efd)


Then, try to start PostgreSQL

**Comment by bloodykheeng** [comment]
---

I was having trouble installing **PostgreSQL** on Windows — the installation kept failing, and I couldn’t figure out why.

Later, I suspected something was already using the default PostgreSQL port, **5432**. To check this, I opened **Command Prompt as Administrator** (this is important for certain commands to work properly) and ran:

```bash
netstat -aon | findstr :5432
```

Output:

```
C:\Users\BK>netstat -aon | findstr :5432
  TCP    0.0.0.0:5432           0.0.0.0:0              LISTENING       6176
  TCP    [::]:5432              [::]:0                 LISTENING       6176
```
> This shows that **port 5432 is in use by a process with PID `6176`**, listening on all network interfaces.

To identify what was using that PID, I ran:

```bash
tasklist /FI "PID eq 6176"
```

Which returned:

```
Image Name                     PID         Session Name        Mem Usage
=========================     ========    ================    ============
postgres.exe                  6176        Console             25,000 K
```

So PostgreSQL was already running in the background and using port 5432.

To fix the issue, I did the following:

1. **Opened Command Prompt as Administrator** (⚠️ required to kill the process successfully).
2. Stopped the process using:

   ```bash
   taskkill /PID 6176 /F
   ```
3. Deleted the existing PostgreSQL data folder to clean up any broken or incomplete setup:

   ```
   C:\laragon\data\postgresql
   ```
4. Restarted **Laragon**, and everything worked perfectly — PostgreSQL started without issues.

---
